### PR TITLE
Space Hogs

### PR DESCRIPTION
**Pull request by Kurfursten**
initial progress for #32699
I won't say "closes" because it's not everything.

Includes
* [x] 5 variants - normal, greased, overgreased, mama, piglet
* [x] Skittishness behavior (flee entirely when full, hit and run when hungry, otherwise full on attack; piglets always flee)
* [x] Mama can spawn piglets (10% chance per life tick if not hungry, costs 15 ticks of nutrition)
* [x] Greased and overgreased hogs can move through airlocks
* [x] Overgreased hogs leave 5 seconds of lube behind them
* [x] All hogs can move through space
* [x] Hogs can be trapped by beartraps
* [x] Uses existing pig sprites
* [x] Different sprites for different pig types
* [x] How pigs GAIN nutrition (handle eating dead bodies, food on floor)
* [x] Piglets nearby increases hostility even if full
* [x] Increased hostility when caged, beartrapped, cannot reach target, wants to retreat at least to distance 2 but can't
* [x] Special headbutt attack on cooldown for kidnapping (spell?)
* [x] Pig scream on death
* [x] Greased variants throw meat when butchered like lube

Not In Scope of this PR
* Elaborate pathing / nesting / search and capture AI
* How these things appear outside of adminbus
* How pigs evolve between states
* Hogs devolve into peaceful piglets on chance with eating cabbage
* Midround ruleset

🆑 
* rscadd: Feral spacehogs have been spotted in this sector.

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/hog.dm [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+	maxHealth = 40
+	health = 40
+	size = SIZE_SMALL
+	retreat_distance = 8 //Retreats and does not approach when it sees a hostile
+	minimum_distance = 8
+
+/mob/living/simple_animal/hostile/spacehog/Process_Spacemove(var/check_drift = 0)
+	return 1 //All spacehogs are proficient in space navigation
+
+/mob/living/simple_animal/hostile/spacehog/Life()
```
Comment (d3athrow) [inline code/modules/mob/living/simple_animal/hostile/hog.dm]: is the two instances of `/mob/living/simple_animal/hostile/spacehog/Life()` intentional

Comment (Kurfursten) [reply code/modules/mob/living/simple_animal/hostile/hog.dm]: Thanks for catching this, the first one is just for mama pigs

File: code/modules/mob/living/simple_animal/hostile/hog.dm [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+/mob/living/simple_animal/hostile/spacehog/mama
+	name = "mama hog"
+	desc = "Manslaughter."
+	speed = 2.5
+	maxHealth = 450
+	health = 450
+	size = SIZE_HUGE
+	nutrition = HOG_MAX
+
+/mob/living/simple_animal/hostile/spacehog/Life()
```
Comment (Kurfursten) [inline code/modules/mob/living/simple_animal/hostile/hog.dm]: this is the one that needs to be bumped down a level to spacehog/mama

Comment (Kurfursten) [reply code/modules/mob/living/simple_animal/hostile/hog.dm:57]: also note to self: bring over death squeal from box

File: code/modules/mob/living/simple_animal/hostile/hog.dm [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+	//Before departing
+	var/turf/simulated/T = loc
+	if(istype(loc) && !T.is_wet())
+		new /obj/effect/overlay/puddle(loc, TURF_WET_LUBE, 5 SECONDS) //leave 5 seconds of lube behind
+	..() //move on
+
+/mob/living/simple_animal/hostile/spacehog/piglet
+	name = "feral space piglet"
+	desc = "This one isn't old enough for grease."
+	icon_state = "pig_1"
```
Comment (Kurfursten) [inline code/modules/mob/living/simple_animal/hostile/hog.dm]: 2 is actually the smallest

**Comment by BomberBro** [comment]
Is this solely AI or could players perhaps play as the hogs (with a timer or something of course)

**Comment by Kurfursten** [comment]
This PR only includes some basic combat behaviors (which won't interfere with player control). Players can be safely be bussed into the hogs.

I told Kernikov I'm not signed on to implement elaborate AI for nesting and kidnapping, but I am willing to set up a midround gamemode that recruits players to be hogs, like how xenos work.

**Comment by Eneocho** [comment]
I'd say it steps a bit too close into the Grue, Xenomorph and Spider territory but I guess it can't hurt to have more variety.

**Comment by BomberBro** [comment]
Yeah I think it'd make for a decent midround ruleset (maybe) if players could join in, not sure how you'd balance it though as I'm not sure how fast hogs spawn or their strength.

**Comment by Brotanics** [comment]
How can I kill 30-50 of them in 3-5 minutes.

**Comment by Eneocho** [comment]
> How can I kill 30-50 of them in 3-5 minutes.

High yield explosives.

**Comment by kernikov** [comment]
I haven't found a spriter yet who can breathe life into the hogkin

**Comment by Theophrastus1** [comment]
The top ten greased up hogs
and the top ten greased up sows

**Comment by kernikov** [comment]
E7EOehfXoAAM-6T

**Comment by TurboGay** [comment]
;HOG

**Comment by Eneocho** [comment]
;HELP THE HOS'S MOTHER IS RAMPAGING IN MEDICAL

**Comment by Blithering** [comment]
I do like the idea that people will soon be yelling "HOS HELP HOG IN HOP HALL"

**Comment by kernikov** [comment]
tfw no hogs until pigs fly

**Comment by Theophrastus1** [comment]
hogs will be soon brother, have faith

**Comment by Kurfursten** [comment]
I'm pushing the content I have now. These are semi-functional but some of their features don't work.
* The headbutt spell works fine when AI is using it, but I can't get it to work for players. The spellcode has been very uncooperative and I have no idea what is wrong with it. If someone can get an idea in code review what is wrong I would be happy to fix it.
* The adults don't seem to follow any of their complex food hunting or retreat behaviors at all and just act like normal hostile mobs (they also don't understand they can move through closed doors when pathfinding). The piglets do always run away as intended though.

As for the rest, it all works. It's technically mergeable. I would like for it to be done feature-complete, but at this point it has been almost 6 months and it needs to be released in some form.

File: code/modules/mob/living/simple_animal/hostile/hog.dm [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+
+/mob/living/simple_animal/hostile/spacehog/proc/idle_search()
+	var/tree_tray = null
+	for(var/obj/machinery/portable_atmospherics/hydroponics/H in view(food_search_radius,src))
+		if(H.seed)
+			if(H.seed.ligneous)
+				tree_tray = H
+			else
+				return H
+	for(var/obj/item/weapon/reagent_containers/food/snacks/meat/M in view(food_search_radius,src))
```
Comment (ShiftyRail) [inline code/modules/mob/living/simple_animal/hostile/hog.dm:145]: Bit weird to call view() so many times but it's been optimised quite a lot so probably ok

**Review by ShiftyRail — APPROVED** [review]
Sorry for leaving this hanging.

**Comment by ExpulsionSnooper** [comment]
HOGBROS

**Pull request merged**